### PR TITLE
Feature/geoprocessing 41 consistent wkt naming

### LINKED ISSUE #41: Improve consistency of terminology around WKT projections
Pygeoprocessing's API is quite consistent about how we name parameters and variables.  One notable exception about this, however, is that `get_raster_info()`'s returned dictionary includes the projection WKT string in a way that is inconsistent with the rest of the pygeoprocessing API.

* `pygeoprocessing.reproject_vector` accepts a parameter `target_wkt`
* `pygeoprocessing.warp_raster` accepts parameters `base_sr_wkt` and `target_sr_wkt`
* `pygeoprocessing.transform_bounding_box` accepts parameters `base_ref_wkt` and `target_ref_wkt`
* `pygeoprocessing.align_and_resize_rasster_stack` accepts `base_sr_wkt_list` and `target_sr_wkt`

At a minimum, `get_raster_info` should use `projection_wkt` as its key.

But perhaps we should also consider making the other WKT-related parameters more consistent.

CCing @richpsharp because we had talked about this in the office

### PR DESCRIPTION
**Pull request by richpsharp**
This changes all the wkt based projection parameters in pygeoprocessing to have a consistent naming schema as described in issue #41. This PR updates the code, tests, and notes in history. Since changing parameter names is not backwards compatible, this branch should be merged into release/2.0

Closes issue #41

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -26,6 +26,11 @@ Unreleased Changes (master)
   might be included with the source distribution.
 * Added ``set_tol_to_zero`` to ``convolve_2d`` to allow for in-function masking
   of near-zero results to be set to 0.0.
+* Renamed all parameters involving Spatial Projectsions to the form
```
Comment (dcdenu4) [inline HISTORY.rst]: typo: Projectsions -> Projections

File: src/pygeoprocessing/geoprocessing.py [hunk 2]
```diff
@@ -719,11 +721,12 @@ def align_and_resize_raster_stack(
                 '"mask_vector_path": %s', vector_mask_options)
         mask_vector_info = get_vector_info(
             vector_mask_options['mask_vector_path'])
-        mask_vector_sr_wkt = mask_vector_info['projection']
-        if mask_vector_sr_wkt is not None and target_sr_wkt is not None:
+        mask_vector_sr_wkt = mask_vector_info['projection_wkt']
+        if mask_vector_sr_wkt is not None and \
```
Comment (dcdenu4) [inline src/pygeoprocessing/geoprocessing.py]: Should `mask_vector_sr_wkt` be `mask_vector_projection_wkt`?

**Review by dcdenu4 — CHANGES_REQUESTED** [review]
Hey @richpsharp just had a few minor suggestions.

File: HISTORY.rst (hunk 1 above)
Comment (richpsharp) [reply HISTORY.rst]: patched in 5eedf44d06fe3765303952cad6b0a2db435ab5c0

File: src/pygeoprocessing/geoprocessing.py (hunk 2 above)
Comment (richpsharp) [reply src/pygeoprocessing/geoprocessing.py]: Patched this and all other internal variable names that use `..sr_wkt` to `..projection_wkt`

**Comment by richpsharp** [comment]
Hey @dcdenu4, I patched those two issues, and I hadn't considered the internal variable names that used projections but I did now and I patched all of them. Also resolved a HISTORY.rst merge conflict.

**Comment by dcdenu4** [comment]
Hey @richpsharp I'm having trouble locating those updates... I see the merge for fixing History.rst conflict and it looks like maybe the typo was preserved during that merge... But the PR `conversation` isn't showing your other commit revisions like: `5eedf44d06fe3765303952cad6b0a2db435ab5c0`. 

It's totally possible I'm just missing something.

**Comment by richpsharp** [comment]
> Hey @richpsharp I'm having trouble locating those updates... I see the merge for fixing History.rst conflict and it looks like maybe the typo was preserved during that merge... But the PR `conversation` isn't showing your other commit revisions like: `5eedf44d06fe3765303952cad6b0a2db435ab5c0`.
> 
> It's totally possible I'm just missing something.

hi @dcdenu4, is it possible I didn't push? I just resolved a merge conflict on HISTORY.rst and I didn't check before but I see the commits I made in the "commits" tab. Can you check it again now? I think it's right...

**Review by dcdenu4 — APPROVED** [review]

**Pull request merged**